### PR TITLE
Configure FDB_API_VERSION dynamically

### LINKED ISSUE #34: Support newer FDB API versions by dynamically detecting C client library version
We currently hardcode API version 620. Users can select an older version through the `api_version` .app environment variable, but newer versions are not allowed. The latest binary release is at 630, and the upcoming 7.0 release will be either 700 or 710.

The tricky bit is that if we hardcode a higher api version, erlfdb will fail to build if a user has an older client library installed. We should have a `./configure` type of interrogation to see what version of the client library is installed, and dynamically configure `FDB_API_VERSION` during the build to match.

The CI system should test for 630 compatibility, but I don't think we should bother with unreleased versions at this time.

More discussion on this topic in the [FDB forums](https://forums.foundationdb.org/t/discovering-max-api-version-when-building-out-of-tree-bindings/2972/6).

### LINKED ISSUE #38: Fully support FDB API version 630
It seems there are a handful of changes in API version 630 that we ought to be accounting for:

https://apple.github.io/foundationdb/api-version-upgrade-guide.html#api-version-630

Specifically:

*  The `get_addresses_for_key` function now returns strings that include the port in the address. Prior to API version 630, this required using the `INCLUDE_PORT_IN_ADDRESS` option, which has now been deprecated.
*  The `ENABLE_SLOW_TASK_PROFILING` network option has been replaced by `ENABLE_RUN_LOOP_PROFILING` and is now deprecated.
* The `FDBKeyValue` struct’s `key` and `value` members have changed type from `void*` to `uint8_t*`.

### PR DESCRIPTION
**Pull request by kocolosk**
This patch set dynamically detects the maximum API version supported by the installed FDB client library and builds the NIF wrapper against that version automatically.

Older API versions can still be selected at runtime as usual through the application environment; this just ensures we have the largest possible range at our disposal, without restricting the versions of the client library we can support with a single branch of this codebase.

Closes #34 
Closes #38

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -0,0 +1,34 @@
+% Hacky means to extract API version from fdbcli protocol version output
+% See https://github.com/apple/foundationdb/blob/master/flow/ProtocolVersion.h
+MaxAPIVersion =
+    begin
+        VsnInfo = os:cmd("fdbcli --version"),
+        {match, [ProtocolStr]} = re:run(VsnInfo, "protocol ([a-f0-9]*)", [{capture, [1], list}]),
+        ProtocolVsn = list_to_integer(ProtocolStr, 16),
+        APIVersionBytes = (ProtocolVsn band 16#0000000FFF00000) bsr 20,
```
Comment (nickva) [inline rebar.config.script:8]: That's quite neat!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,6 +8,9 @@ on:
 jobs:
 
   build_on_linux:
+    strategy:
+      matrix:
+        fdb-version: ['6.1.13', '6.2.30', '6.3.22']
```
Comment (nickva) [inline .github/workflows/ci.yml:13]: 620 might be the lowest we were configured to go. I saw some tests fail with it. If we want to support it we might have ifdef a bunch of API functions and arguments.

**Review by nickva — APPROVED** [review]
A very nice improvement

There is a note about 610 support but otherwise if the build passes +1

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kocolosk) [reply .github/workflows/ci.yml]: You're quite right, I found your work in c89b2d6e and figured supporting 6.1.x can be left for the future if anyone ever wants that support :)

File: rebar.config.script (hunk 1 above)
Comment (kocolosk) [reply rebar.config.script:8]: Hah, I thought it seemed quite hacky, but looking at the way it's defined in FDB I suspect this ends up being pretty reliable

**Comment by kocolosk** [comment]
While we are technically green on 6.3 here, it seems there are a handful of changes in API version 630 that we ought to be accounting for:

https://apple.github.io/foundationdb/api-version-upgrade-guide.html#api-version-630

Specifically:

*  The `get_addresses_for_key` function now returns strings that include the port in the address. Prior to API version 630, this required using the `INCLUDE_PORT_IN_ADDRESS` option, which has now been deprecated.
*  The `ENABLE_SLOW_TASK_PROFILING` network option has been replaced by `ENABLE_RUN_LOOP_PROFILING` and is now deprecated.
* The `FDBKeyValue` struct’s `key` and `value` members have changed type from `void*` to `uint8_t*`.

**Comment by kocolosk** [comment]
Will squash the CI commits into one to clean up the history, but this is stacked on top of #33 so I think I may wait until that one is merged to proceed further.

**Pull request merged**